### PR TITLE
(Deposit/Withdraw) Support Nomic Withdrawals

### PR DESCRIPTION
**Pull request by JoseRFelix**
## What is the purpose of the change:

To enable Bitcoin withdrawals, update the Nomic provider to support withdrawals via IBC forwarding to Nomic. For alloyed BTC, the process will involve swapping it to nBTC before initiating the forwarding.

### Linear Task

https://linear.app/osmosis/issue/FE-1140/support-withdrawals

## Brief Changelog

- Add `getQuote` implementation to Nomic provider
- Move swap messages to the `tx` package to improve reusability and modularity
- Create swap router complex queries to fetch quotes from providers like Nomic

## Testing and Verifying

- [ ] Can transfer IBC assets
- [ ] Can swap given in or out 
- [ ] Can withdraw alloyed BTC to Bitcoin network
- [ ] Can withdraw nomic BTC to Bitcoin network
- [ ] Can Deposit Nomic BTC and see the pending deposit

### CODE REVIEW

File: packages/bridge/src/nomic/index.ts [hunk 1]
```diff
@@ -36,12 +52,224 @@ export class NomicBridgeProvider implements BridgeProvider {
...
+        desiredBech32Prefix: "nomic",
+      }),
+    };
+
+    const [ibcTxMessages, estimatedTime] = await Promise.all([
+      ibcProvider.getTxMessages({
+        ...transactionDataParams,
+        memo: destMemo,
+      }),
+      ibcProvider.estimateTransferTime(
```
Comment (jonator) [inline packages/bridge/src/nomic/index.ts:175]: Does the estimated time include confirmation on Bitcoin network side? Maybe we need to hardcode something for that. IDK how nomic withdrawals work though

File: packages/bridge/src/nomic/index.ts [hunk 2]
```diff
@@ -12,18 +24,22 @@ import {
...
   GetDepositAddressParams,
 } from "../interface";
+import { getGasAsset } from "../utils/gas";
 
 export class NomicBridgeProvider implements BridgeProvider {
   static readonly ID = "Nomic";
   readonly providerName = NomicBridgeProvider.ID;
 
   relayers: string[];
   nBTCMinimalDenom: string;
```
Comment (jonator) [inline packages/bridge/src/nomic/index.ts:41]: nit: maybe this should be `readonly` just to be safe

File: packages/bridge/src/nomic/index.ts [hunk 3]
```diff
@@ -228,4 +471,19 @@ export class NomicBridgeProvider implements BridgeProvider {
       return [];
     }
   }
+
+  async getProtoRegistry() {
+    if (!this.protoRegistry) {
+      const [{ ibcProtoRegistry, osmosisProtoRegistry }, { Registry }] =
+        await Promise.all([
```
Comment (jonator) [inline packages/bridge/src/nomic/index.ts:478]: Is this valuable on the server? More of a Q vs a request. Does it slightly reduce cold start times by doing something like this?

File: packages/bridge/src/utils/gas.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    return {
+      address: feeCurrency.chainSuggestionDenom,
+      denom: feeCurrency.coinDenom,
+      decimals: feeCurrency.coinDecimals,
+      coinGeckoId: feeCurrency.coinGeckoId,
+    };
+  }
+}
+
+export function getChains({
```
Comment (jonator) [inline packages/bridge/src/utils/gas.ts:57]: Maybe it would be helpful to comment where these chains come from

File: packages/bridge/src/interface.ts [hunk 5]
```diff
@@ -223,6 +224,13 @@ export const bridgeAssetSchema = z.object({
 
 export type BridgeAsset = z.infer<typeof bridgeAssetSchema>;
 
+// TODO: Add comment explaining this
```
Comment (jonator) [inline packages/bridge/src/interface.ts:227]: Cleanup

File: packages/server/src/queries/complex/swap.ts [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+  tokenInAmount: z.string(),
+  tokenOutDenom: z.string(),
+  forcePoolId: z.string().optional(),
+});
+
+type GetRouteTokenOutGivenInParams = z.infer<
+  typeof getRouteTokenOutGivenInParams
+>;
+
+export async function getRouteTokenOutGivenIn({
```
Comment (jonator) [inline packages/server/src/queries/complex/swap.ts:21]: Are these driveby changes? May want to work coordinate w @mattupham, may be conflicts here

File: packages/tx/src/message-composers/complex/swap.ts [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+
+    return {
+      routes,
+      tokenOut,
+      tokenInMaxAmount,
+    };
+  }
+}
+
+export async function getSwapMessages({
```
Comment (jonator) [inline packages/tx/src/message-composers/complex/swap.ts:151]: I think @mattupham is also using this. Good we organized/moved them here along with other composers. Cleaner!

File: packages/web/modals/review-order.tsx [hunk 8]
```diff
@@ -27,7 +28,7 @@ import {
   useWindowSize,
 } from "~/hooks";
 import { isValidNumericalRawInput } from "~/hooks/input/use-amount-input";
-import { QuoteType, useSwap } from "~/hooks/use-swap";
```
Comment (jonator) [inline packages/web/modals/review-order.tsx:30]: Nit: I prefer something more specific like `Quote` over `QuoteType`. We know it's a type.

**Review by jonator — APPROVED** [review]

**Review by jonator — APPROVED** [review]
LGTM, w some comments

File: packages/bridge/src/nomic/index.ts (hunk 1 above)
Comment (JoseRFelix) [reply packages/bridge/src/nomic/index.ts:175]: Yeah, I’m coordinating with Nomic to figure out how we can track the progress and status of withdrawals. For now, I think it’s best to set the estimated time based on IBC, as that’s the only timeframe we are certain about.

File: packages/server/src/queries/complex/swap.ts (hunk 6 above)
Comment (JoseRFelix) [reply packages/server/src/queries/complex/swap.ts:21]: @mattupham This [comment](https://github.com/osmosis-labs/osmosis-frontend/pull/3885#discussion_r1805490532) should resolve this issue.

File: packages/bridge/src/nomic/index.ts (hunk 3 above)
Comment (JoseRFelix) [reply packages/bridge/src/nomic/index.ts:478]: I'm not entirely sure. I'll have to investigate
